packages/amplify-graphql-api-construct-tests/src/graphql-request.ts

```diff
@@ -5,24 +5,13 @@ export type GraphqlResponse = {
   body: any;
 };
 
-export const graphql = async (apiEndpoint: string, apiKey: string, query: string): Promise<GraphqlResponse> => {
-  const options = {
-    method: 'POST',
-    headers: {
-      'x-api-key': apiKey,
-      'Content-Type': 'application/json',
-    },
-    body: JSON.stringify({ query }),
-  };
-
-  const request = new Request(apiEndpoint, options);
-
+const graphqlRequest = async (apiEndpoint: string, payload: any): Promise<GraphqlResponse> => {
   let statusCode = 200;
   let body;
   let response;
 
   try {
-    response = await fetch(request);
+    response = await fetch(new Request(apiEndpoint, payload));
     body = await response.json();
     if (body.errors) statusCode = 400;
```

Comment:
(Side note: I know this is pre-existing code, but it seems odd that we're overriding the status code from the service. I recognize that the service returning 200 in error conditions is weird behavior driven by the need to support partial responses, but it's strange that our tests suppress that behavior rather than dealing with it.)

Reply:
I agree, this took me a little while to figure out why the response code wasn't what I expected. I don't want to go through the 2-4 hour cycle for running e2e's again, so I'll track that as a fast-follow today.